BugsnagRequestMonitor/BugsnagRequestMonitorTests/BugsnagRequestMonitorTests.m

```diff
@@ -5,15 +5,535 @@
 //  Created by Karl Stenerud on 26.08.21.
 //
 
+#import "BSGURLSessionTracingDelegate.h"
 #import <XCTest/XCTest.h>
 
-@interface BugsnagRequestMonitorTests : XCTestCase
+#define ONE_SECOND_IN_NS 1000000000ULL
 
+#pragma mark MockURLProtocol
+
+@interface MockURLProtocol: NSURLProtocol
+@end
+
+@implementation MockURLProtocol
+
+static NSInteger mock_nextStatusCode;
+static NSData *mock_nextData;
+
++ (BOOL)canInitWithRequest:(NSURLRequest *)request {
+    return YES;
+}
+
++ (NSURLRequest *)canonicalRequestForRequest:(NSURLRequest *)request {
+    return request;
+}
+
+- (void)stopLoading {
+}
+
+- (void)startLoading {
+    NSHTTPURLResponse *response = [[NSHTTPURLResponse alloc] initWithURL:self.request.URL statusCode:mock_nextStatusCode HTTPVersion:@"HTTP/1.1" headerFields:@{@"Content-Length": [NSString stringWithFormat:@"%lu", (unsigned long)mock_nextData.length]}];
+
+    [self.client URLProtocol:self didReceiveResponse:response cacheStoragePolicy:NSURLCacheStorageNotAllowed];
+    if (mock_nextData) {
+        [self.client URLProtocol:self didLoadData:mock_nextData];
+    }
+    [self.client URLProtocolDidFinishLoading:self];
+}
+
++ (void)setNextStatusCode:(NSInteger)statusCode data:(NSData *)data {
+    mock_nextStatusCode = statusCode;
+    mock_nextData = data;
+}
+
+@end
+
+#pragma mark BugsnagRequestMonitorTests
+
+@interface BugsnagRequestMonitorTests : XCTestCase <BSGBreadcrumbSink>
```

Comment:
There don't seem to be any tests cases that cover a non-zero reqLength?

Reply:
There was, but it wasn't triggering because `transaction.request` seems to come with a zero-length body and the test hadn't had its length requirement updated. Test now requires the correct length, and intercept code now uses `task.originalRequest`.

https://github.com/bugsnag/bugsnag-cocoa/pull/1181/files#diff-ab26956c0f7f778219f804a6f357de4ebb1e1aa58890b84ac1721024b9b928fdR466